src/video-player/components/runtime.tsx

```diff
@@ -116,32 +128,45 @@ export const Runtime: React.FC<IProps> = ({ authoredState, interactiveState, set
 
   const handlePlaying = () => {
     setHasStartedPlayback(true);
-    // store current playback progress each second
-    if (Math.trunc(getViewTime()) > saveStateInterval.current) {
-      saveStateInterval.current += 1;
-      updateState();
+    if (captionsTrackRef.current) {
+      setCaptionDisplayState(captionsTrackRef.current.mode);
     }
+    updateState();
   };
 
   const handlePlay = () => {
-    const percentageViewed = getViewPercentage();
-    log("video started", { videoUrl: authoredState.videoUrl, percentage_viewed: percentageViewed });
+    updateState(true, "video started");
+  };
+
+  const handleSeek = () => {
+    updateState(true, "video time scrubber used");
   };
 
   const handleStop = () => {
-    updateState();
-    const percentageViewed = getViewPercentage();
-    log("video stopped", { video_url: authoredState.videoUrl, percentage_viewed: percentageViewed });
+    updateState(true, "video stopped");
   };
 
-  const updateState = () => {
-    setInteractiveState?.(prevState => ({
-      ...prevState,
-      answerType: "interactive_state",
-      answerText: `Percentage viewed: ${viewedProgress}`,
-      percentageViewed: getViewPercentage(),
-      lastViewedTimestamp: getViewTime()
-    }));
+  const updateState = (forceSave = false, logMessage = "") => {
+    // store current playback progress each second
+    const updateIntervalReady = Math.trunc(getViewTime()) > saveStateInterval.current;
+    if (updateIntervalReady || forceSave) {
+      const percentageViewed = getViewPercentage();
+      saveStateInterval.current += 1;
+      setInteractiveState?.(prevState => ({
+        ...prevState,
+        answerType: "interactive_state",
+        answerText: `Percentage viewed: ${viewedProgress}`,
+        percentageViewed,
+        lastViewedTimestamp: getViewTime()
+      }));
+      if (logMessage.length > 0) {
```

Comment:
This reads slightly confusing to me, as I normally associate `.length` with array-like objects. Any reason not to have just `if (logMessage)` which would come to the same thing? And, indeed, the logMessage argument could be optional without the `""` default, (`logMessage: string?`), which would come to the same thing and be even clearer (to my eyes).